src/object.di

```diff
@@ -595,6 +595,82 @@ inout(V) get(K, V)(inout(V[K])* aa, K key, lazy inout(V) defaultValue)
     return (*aa).get(key, defaultValue);
 }
 
+private void _destructRecurse(S)(ref S s)
+    if (is(S == struct))
+{
+    import core.internal.traits : hasElaborateDestructor;
+
+    static if (__traits(hasMember, S, "__dtor"))
+        s.__dtor();
+
+    foreach_reverse (i, ref field; s.tupleof)
+    {
+        static if (hasElaborateDestructor!(typeof(field)))
+            _destructRecurse(field);
+    }
+}
+
+private void _destructRecurse(E, size_t n)(ref E[n] arr)
+{
+    import core.internal.traits : hasElaborateDestructor;
+
+    static if (hasElaborateDestructor!E)
+    {
+        foreach_reverse (ref elem; arr)
+            _destructRecurse(elem);
+    }
+}
+
+private string _genFieldPostblit(S)()
+{
+    import core.internal.traits : hasElaborateCopyConstructor;
+    import core.internal.traits : FieldNameTuple;
+
+    string code;
+    foreach(fieldName; FieldNameTuple!S)
+    {
+        static if(hasElaborateCopyConstructor!(typeof(__traits(getMember, S.init, fieldName))))
+        {
+            code ~= `
+                _postblitRecurse(s.` ~ fieldName ~ `);
+                scope(failure) _destructRecurse(s. ` ~ fieldName ~ `);
```

Comment:
I haven't tested anything for performance/bloat. Still need to check the array postblit code, too.

> Maybe it warrants testing if the postblit can throw at all? Unfortunately, the test would require `functionAttributes`...

That would be clever. The compiler has historically been pretty bad at optimizing `scope(...)` statements in my experience, like inserting a bunch of try-catch code for `scope(success)`.